code/script.js

```diff
@@ -1,19 +1,131 @@
 // Start here
 
+const mainType = ["Pizza", "Pasta", "Salad"];
+
+const pizzaType = ["Napolitana", "Hawaian", "Pepperoni"];
+const pastaType = [
+  "Spaghetti Carbonara",
+  "Fettuccine Alfredo",
+  "Cheesy Tortellini",
+];
+const saladType = ["Caesar Salad", "Caprese Salad", "Greek Salad"];
+
+const angryChief =
+  "The chief gets angry: You aren't choosing carefully! No food for you!";
+
 // Step 1 - Welcome and introduction
 // Your code goes here
 alert(
   `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
-)
+);
+
+let userName = prompt("What's your name?");
+
+if (userName.length === 0) {
+  alert(`Hello, stranger!`);
+} else {
+  alert(`Hi, ${userName}!`);
+}
 
 // Step 2 - Food choice
 // Your code goes here
+let userInput = prompt(
+  `What type of food would you like to order?Enter a number:
+     1 - ${mainType[0]}
+     2 - ${mainType[1]}
+     3 - ${mainType[2]}`
+);
+
+let orderNumber = parseInt(userInput);
+let mealType;
+const mainTypeIndex = orderNumber - 1;
+
+if (orderNumber > 0 && orderNumber <= mainType.length) {
+  mealType = mainType[mainTypeIndex];
+  alert(`You've chosen ${mealType}!`);
+} else {
+  alert(`${angryChief}`);
+}
 
 // Step 3 - Subtype choice
 // Your code goes here
+let subTypeArray;
+
+switch (orderNumber) {
+  case 1:
+    subTypeArray = pizzaType;
+    break;
+
+  case 2:
+    subTypeArray = pastaType;
+    break;
+
+  case 3:
+    subTypeArray = saladType;
+    break;
+
+  default:
+    alert(`${angryChief}`);
+    break;
+}
+
+let subInput = prompt(
```

Comment:
as an improvement, you can add a default case for `subInput` in case of invalid input, like you did previously 👍